gen.go

```diff
@@ -17,15 +18,26 @@ const MaxLength = 8192
 
 const ByteArrayMaxLen = 2 << 20
 
-const MaxLenTag = "maxlen"
-const NoUsrMaxLen = -1
+const (
+	MaxLenTag   = "maxlen"
+	NoUsrMaxLen = -1
+)
 
 var (
 	cidType      = reflect.TypeOf(cid.Cid{})
 	bigIntType   = reflect.TypeOf(big.Int{})
 	deferredType = reflect.TypeOf(Deferred{})
 )
 
+// CborSerializer is a base interface for CBOR serializers that can be used as
+// a generic type T such that a serialization and deserialization methods
+// will be called on any field of type T.
+type CborSerializer[V any] interface {
+	New() V
+	MarshalCBOR(w io.Writer) error
+	UnmarshalCBOR(r io.Reader) error
+}
```

Comment:
nit: I'd only handle the unmarshal side here.

```suggestion
type GenericCBORUnmarshaler[V any] interface {
	New() V
	CBORUnmarshaler
}
```

Also, I'm wondering if we want to make this:

```suggestion
type GenericCBORUnmarshaler[V any] interface {
	NewFromCBOR(r io.Reader) V
}
```

Not sure... it's kind of funky either way.

Reply:
Nah, let's stick with the original `New()`. Then add a helper to make it less strange.

Reply:
why not `MarshalCBOR`? we need to be able to call down into it for a generic type.

this interface isn't binding here in the code anywhere, it's intended to be used like we do with `HamtValue`, but over there we're going to add `Equals` to this.

Reply:
I guess I like to be able to have some types that only support "viewing" (and it would be nice to mirror the current interfaces). But you're right that's useful to have a combined version and we don't actually _use_ the constraint here.